src/main/java/com/zendesk/maxwell/replay/ReplayBinlogStore.java

```diff
@@ -0,0 +1,138 @@
+package com.zendesk.maxwell.replay;
+
+import com.zendesk.maxwell.CaseSensitivity;
+import com.zendesk.maxwell.filtering.Filter;
+import com.zendesk.maxwell.replication.Position;
+import com.zendesk.maxwell.schema.*;
+import com.zendesk.maxwell.schema.ddl.*;
+import com.zendesk.maxwell.util.ConnectionPool;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author udyr@shlaji.com
+ */
+public class ReplayBinlogStore extends AbstractSchemaStore implements SchemaStore {
+	private static final Logger LOGGER = LoggerFactory.getLogger(com.zendesk.maxwell.replay.ReplayBinlogStore.class);
+
+	private final ConnectionPool schemaConnectionPool;
+	private final CaseSensitivity caseSensitivity;
+	private final Filter filter;
+	private Schema maxwellOnlySchema;
+	private ReplayConfig config;
+
+	public ReplayBinlogStore(ConnectionPool schemaConnectionPool, CaseSensitivity caseSensitivity, ReplayConfig config) {
+		super(schemaConnectionPool, schemaConnectionPool, caseSensitivity, config.filter);
+		this.filter = config.filter;
+		this.schemaConnectionPool = schemaConnectionPool;
+		this.caseSensitivity = caseSensitivity;
+		this.config = config;
+	}
+
+	@Override
+	public Schema getSchema() throws SchemaStoreException {
+		if (maxwellOnlySchema != null) {
+			return maxwellOnlySchema;
+		}
+
+		synchronized (com.zendesk.maxwell.replay.ReplayBinlogStore.class) {
+			if (maxwellOnlySchema != null) {
+				return maxwellOnlySchema;
+			}
+
+			try (Connection conn = schemaConnectionPool.getConnection()) {
+				SchemaCapturer capturer = new SchemaCapturer(conn, caseSensitivity);
+				maxwellOnlySchema = capturer.capture();
```

Comment:
so this kinda sucks -- basically it means that the binlog files are only replayable as long as the schema hasn't changed.  at the very least we'd have to document this pretty severe limitation.  There is a somewhat better method using binlog heartbeats to try to figure out what schema to start with, but it's a LOT more complicated.

Reply:
Keeping the schema consistent with Binlog is difficult unless the mysql service is initialized using Maxwell without interruption. In addition, I used max_schemas=1000, which will cause the schema to be cleared

Reply:
I use mysqldump to back up the schema every day. When I need to replay binlog, I will restore the backup schema to a temporary mysql service and provide it to Maxwell to read the correct schema

Reply:
I used max_schemas =1000, which will cause the schema to be cleared. So I can't use heartbeats.

Using the correct schema and position, chase binlog from this positionto the desired output position.

This schema needs to be provided by the user. If you use Maxwell's schema, it may be missing.

The user didn't keep Maxwell running healthily all the time, and it may have been restarted. Maybe Maxwell has restored the latest schema

The **replay** does not need whether Maxwell has run before, but only needs a correct schema and all binlogs. Does this make the **replay** more simple and reliable?

Reply:
> The replay does not need whether Maxwell has run before, but only needs a correct schema and all binlogs. Does this make the replay more simple and reliable?

it makes it simpler from the perspective of this code, but operationally it's terrible: you'd need to somehow have a copy of the schema from the point in time that you're playing the binlogs from -- if the schema has changed at all, you'd need a database backup or something.